yamllint/rules/float_values.py

```diff
@@ -0,0 +1,158 @@
+# Copyright (C) 2022 the yamllint contributors
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Use this rule to limit the permitted values for floating-point numbers.
+YAML permits three classes of float expressions: approximation to real numbers,
+positive and negative infinity and "not a number".
+
+.. rubric:: Options
+
+* Use ``require-numeral-before-decimal'`` to require floats to start
+    with a numeral (ex ``0.0`` instead of ``.0``).
+* Use ``forbid-scientific-notation`` to forbid scientific notation.
+* Use ``forbid-nan`` to forbid NaN values.
```

Comment:
Not everybody knows what NaN means, and it's easy to be explicit in this line, so I propose:

```suggestion
* Use ``forbid-nan`` to forbid NaN (not a number) values.
```